src/frontend/frontend/console.js

```diff
@@ -46,214 +46,113 @@ angular.module('frontend-app')
-            var numFrameObjects = frameinfopatt.exec(line)[1];
-            self._write(sprintf("\n  This frame has %s object(s):\n", numFrameObjects));
-          } else if (framevarpatt.test(line)) {
-            var frameVarInfo = framevarpatt.exec(line);
-            var objectSize = parseInt(frameVarInfo[2]) - parseInt(frameVarInfo[1]);
-            self._write(sprintf("  %d byte object %s located %s bytes into frame.", objectSize, frameVarInfo[3], frameVarInfo[1]));
-            if (/overflow/.test(line)) {
-              self._write("  Access overflew this variable.\n");
-            } else if (/underflow/.test(line)) {
-              self._write("  Access underflew this variable.\n");
-            } else {
-              self._write("\n");
-            }
-          } else if (/^previously allocated/.test(line)) {
-            self._write("\n  Allocated by:\n");
-          } else if (/^freed by/.test(line)) {
-            self._write("freed already by:\n");
-          } else if(suppglobaladdrpatt.test(line)) {
-            var globalAddrInfo = suppglobaladdrpatt.exec(line);
-            var shortAddrFileInfo = globalAddrInfo[5].split('/').pop();
-            self._write(sprintf("\n  %s is %s bytes %s of %s byte(s) global variable %s (located %s) defined at %s:%s:%s.\n",
-                                globalAddrInfo[1],
-                                globalAddrInfo[2],
-                                globalAddrInfo[3],
-                                globalAddrInfo[9],
-                                globalAddrInfo[4],
-                                globalAddrInfo[8],
-                                shortAddrFileInfo,
-                                globalAddrInfo[6],
-                                globalAddrInfo[7]));
-          }
-        } catch (e) {
-          console.log("Could not produce stack trace from line: %s, %s", line, e);
+    function print_asan_error(err) {
+      // json object should have:
+      // error_type (string)
+      // depending on error_type, you can assume each frame-list will have certain key-value pairs, AND
+      // at the top level, the json object passed in will also have certain key-value pairs
+      // **OR**: have a misc field and just iterate over it and print everything (at the same level as frame_list AND/OR at the very top level with
+      // call_stacks: hash (definitely has frame-list; may have other members)
+      // frame_list: list of frames (with line numbers, columns, files, etc.
+      // frame: column, file, frame#, function, function_offset, line, module, offset
+      // raw message
+      maybe_log(err);
+      if(err.error_type == "unknown") { return; }
+      var to_print = [];
+      to_print.push("================================================");
+      to_print.push("Memory error occurred! Type of error: " + err.error_type);
+      for (var current_stack = 0; current_stack < err.call_stacks.length; current_stack++) {
```

Comment:
This can be done later, but your output uses a lot of whitespace right now.  I think something like our current output format would be easier to read.